tensorflow/python/framework/graph_util_impl.py

```diff
@@ -248,7 +248,8 @@ def convert_variables_to_constants(sess,
                                    input_graph_def,
                                    output_node_names,
                                    variable_names_whitelist=None,
-                                   variable_names_blacklist=None):
+                                   variable_names_blacklist=None,
+                                   check_and_revert_if_type_mismatch=False):
```

Comment:
Can we document what this new argument does? Ideally with a usage example. 
perhaps the name could be more self explanatory, right now it's not clear from the name what it does.

Also, when would someone want this to be `False`? If no one should want this to be `False` we can just make the change and not have it as an option.

Reply:
For the usage example, do you want to a test? Or just an simple example? 
The new test i added (testGraphWithResourceScatter) uses this argument.
The idea: After freezing if there are nodes that have a type mismatch, revert the variable and assign on read.

It might not be ideal for certain people to still have Variables in a frozen graph, so this is by default off. As i have also added an import check at the end of the freeze, if the graph fails this check, you can try the check_and_revert argument. 

for the argument name, i can rename to fix_type_mismatch or if you have any better suggestions.

Reply:
@guptapriya Can you please take a look on above comments from @sboshin. Thanks!

Reply:
A test is always good to prevent regressions.

A doctest example is the best. It acts as both documentation and test. See https://www.tensorflow.org/community/contribute/docs_ref

Reply:
@sboshin  Can you please check @mihaimaruseac's comments and keep us posted ? Thanks!